test/fluree/db/ledger/query/analytical_test.clj

```diff
@@ -247,6 +248,38 @@
       (is (= sorted-res res)
           (str "Results were not ordered by ?p" (pr-str res))))))
 
+(deftest order-by-limit-offset
+  (testing "orderBy query"
+    (let [ledger          (test/rand-ledger test/ledger-chat
+                                            {:http/schema ["chat.edn" "chat-preds.edn"]})
+          {:keys [block]} (test/transact-data ledger "chat.edn")
+          db              (fdb/db (:conn test/system) ledger {:syncTo block})
+          query-all       {:select  {"?e" ["person/handle"
+                                           {"comment/_person" {"_as" "comments"}}]}
+                           :where   [["?e" "person/favNums" "?favNums"]]
+                           :orderBy "person/handle"}
+          all-results     (async/<!! (fdb/query-async db query-all))
+          total-count     (count all-results)]
+
+      (testing "with limit"
+        (let [limit       1
+              query-limit (assoc query-all :limit limit)
+              subject     (async/<!! (fdb/query-async db query-limit))]
+
+          (is (= subject (into [] (take limit) all-results))
+              "returns subjects only up to the limit")
+
+          (testing "and with offset"
+            (let [offset       1
+                  query-offset (assoc query-limit :offset offset)
+                  subject      (async/<!! (fdb/query-async db query-offset))]
+
+              (is (= subject (into []
+                                   (comp (drop offset)
+                                         (take limit))
+                                   all-results))
+                  "returns subjects only up to the limit after the offset"))))))))
```

Comment:
I'd have _slightly_ more confidence in these tests if the expected values were either hard-coded values or calculated/compared in a much more primitive way (vs. the logic implementation in the code under test).

E.g. for offest=1, limit=1
```clojure
(is (= 1 (count subject)))
(is (= (first subject) (second all-results)))
```

Otherwise they could be wrong for the same reason(s) and agree with each other and pass.

Not a huge deal here, so I'll leave it up to you.

Reply:
yeah, i definitely had some hesitance about using exactly the same algorithm too. i think i got too clever in the name of test robustness. i'll change it.